sycl/include/CL/sycl/detail/pi.h

```diff
@@ -797,6 +775,22 @@ pi_result piDevicePartition(pi_device device,
                             pi_uint32 num_devices, pi_device *out_devices,
                             pi_uint32 *out_num_devices);
 
+/// Gets the native handle of a PI device object.
+///
+/// \param device is the PI device to get the native handle of.
+/// \param nativeHandle is the native handle of device.
+pi_result piextDeviceGetNative(pi_device device,
+                               pi_native_handle *nativeHandle);
+
+/// Creates PI device object from a native handle.
+/// NOTE: The instance of the PI device created must have a reference count of
+///       one higher than at the time of the call.
```

Comment:
But the PI object didn't exists before the call, what's "one higher" then?

Reply:
I've clarified this in terms of "shared reference counts" just to not make it too OpenCL specific.